quarkusio/quarkus
# Support extension registry when adding or listing extensions in plugins

This PR introduces support for querying the extension registry compiled from https://github.com/quarkusio/quarkus-extension-catalog. 

It will fallback to the original behavior if the registry URL is not specified.

## To test: 

Make sure to change the Quarkus BOM dependency in the project to a released version which is also present in the registry(eg. 1.5.1.Final)

### Maven
```
mvn quarkus:list-extensions -DsearchPattern=kubernetes -Dregistry=https://github.com/quarkusio/quarkus-extension-catalog/releases/download/v23/registry.json
```

### Gradle

```
gradle listExtensions --searchPattern=mutiny --registry=https://github.com/quarkusio/quarkus-extension-catalog/releases/download/v23/registry.json
```

Fixes #9593
